### ISSUE TITLE
strange problem with ParameterNumber during checkstyle build

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#215]
There is build failure 
https://travis-ci.org/checkstyle/checkstyle/jobs/31290616
between on commit:
```
commit 02e6ed40387d10a17db48446e8856bbd644f04da
Author: Roman Ivanov <ivanov-.......>
Date:   Wed Jul 30 17:44:09 2014 -0700

    TODO was removed
```

and there is not build failure on commit:
```
commit 85265a9b287ec860ee5c4b690bd913874d60f03a
Author: Jarmo Isotalo <jamo@..>
Date:   Tue Jun 10 13:43:59 2014 +0300

    Added the missing period
```

Problem is in http://checkstyle.sourceforge.net/config_sizes.html#ParameterNumber
And violation is correct, but it did not failed before comment removal ?

When I single out that Check in separate config (1check.xml) and run against that file, it always fail on both versions of files.
```
java -jar target/checkstyle-5.8-SNAPSHOT-all.jar -c 1check.xml src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java
```

1check.xml
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
  <property name="severity" value="error"/>
  <module name="TreeWalker">
    <module name="ParameterNumber">
      <property name="id" value="paramNum"/>
    </module>
 </module>
</module>
```

for now, I changed a Check config to let build pass.

**Comment by MEZk** [comment]
**checkstyle-6.11-SNAPSHOT. Use of ```mvn clean verify```**
```xml
<module name="ParameterNumber">
  <property name="id" value="paramNum"/>
</module>
```
1) Without changes by commit 02e6ed40387d10a17db48446e8856bbd644f04da and 85265a9b287ec860ee5c4b690bd913874d60f03a
```
[checkstyle] /media/andreiselkin/TOURO/IDEA_Projects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:106:12: More than 7 parameters (found 9).
[checkstyle] /media/andreiselkin/TOURO/IDEA_Projects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:144:12: More than 7 parameters (found 8).
[checkstyle] /media/andreiselkin/TOURO/IDEA_Projects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:175:12: More than 7 parameters (found 8).
[checkstyle] /media/andreiselkin/TOURO/IDEA_Projects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:237: First sentence of Javadoc is incomplete (period is missing) or not present.
[checkstyle] /media/andreiselkin/TOURO/IDEA_Projects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:237: First sentence should end with a period.
```
ParameterNumber works fine.

2) With changes by commit 02e6ed40387d10a17db48446e8856bbd644f04da
```
[checkstyle] Running Checkstyle 6.11-SNAPSHOT on 708 files
[checkstyle] /media/andreiselkin/TOURO/IDEA_Projects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:106:12: More than 7 parameters (found 9).
[checkstyle] /media/andreiselkin/TOURO/IDEA_Projects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:144:12: More than 7 parameters (found 8).
[checkstyle] /media/andreiselkin/TOURO/IDEA_Projects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:175:12: More than 7 parameters (found 8).
[checkstyle] /media/andreiselkin/TOURO/IDEA_Projects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:237: First sentence of Javadoc is incomplete (period is missing) or not present.
[checkstyle] /media/andreiselkin/TOURO/IDEA_Projects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:237: First sentence should end with a period.
```
ParameterNumber works fine.

3) With changes by commit 02e6ed40387d10a17db48446e8856bbd644f04da and 85265a9b287ec860ee5c4b690bd913874d60f03a
```
[checkstyle] /media/andreiselkin/TOURO/IDEA_Projects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:106:12: More than 7 parameters (found 9).
[checkstyle] /media/andreiselkin/TOURO/IDEA_Projects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:144:12: More than 7 parameters (found 8).
[checkstyle] /media/andreiselkin/TOURO/IDEA_Projects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:175:12: More than 7 parameters (found 8).
```
ParameterNumberWorks fine.

4) With changes by commit 85265a9b287ec860ee5c4b690bd913874d60f03a and without changes by commit 02e6ed40387d10a17db48446e8856bbd644f04da
```
[checkstyle] /media/andreiselkin/TOURO/IDEA_Projects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:60: Comment matches to-do format '(TODO)|(FIXME)'.
[checkstyle] /media/andreiselkin/TOURO/IDEA_Projects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:107:12: More than 7 parameters (found 9).
[checkstyle] /media/andreiselkin/TOURO/IDEA_Projects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:145:12: More than 7 parameters (found 8).
[checkstyle] /media/andreiselkin/TOURO/IDEA_Projects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:176:12: More than 7 parameters (found 8).
[checkstyle] /media/andreiselkin/TOURO/IDEA_Projects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:212:5: Cyclomatic Complexity is 12 (max allowed is 10).
```
ParameterNumber works fine, but look at CyclomaticComplexity. It seams, that it is incorrect. Does TODO or increasing of lines in file affects CC? It should not. 

**checkstyle-6.11-SNAPSHOT. Use of checkstyle-checker**
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">
<module name="Checker">
      <module name="TreeWalker">
      <module name="ParameterNumber">
      <property name="id" value="paramNum"/>
    </module>
</module>
</module>
```
```
java -jar checkstyle-6.11-SNAPSHOT-all.jar -c checks.xml LocalizedMessage.java
```
1) Without changes by commit 02e6ed40387d10a17db48446e8856bbd644f04da and 85265a9b287ec860ee5c4b690bd913874d60f03a
```
Starting audit...
/home/andreiselkin/Downloads/temp/LocalizedMessage.java:107:12: More than 7 parameters (found 9).
/home/andreiselkin/Downloads/temp/LocalizedMessage.java:145:12: More than 7 parameters (found 8).
/home/andreiselkin/Downloads/temp/LocalizedMessage.java:176:12: More than 7 parameters (found 8).
Audit done.
Checkstyle ends with 3 errors.
```
2) With changes by commit 02e6ed40387d10a17db48446e8856bbd644f04da
```
Starting audit...
/home/andreiselkin/Downloads/temp/LocalizedMessage.java:106:12: More than 7 parameters (found 9).
/home/andreiselkin/Downloads/temp/LocalizedMessage.java:144:12: More than 7 parameters (found 8).
/home/andreiselkin/Downloads/temp/LocalizedMessage.java:175:12: More than 7 parameters (found 8).
Audit done.
Checkstyle ends with 3 errors.
```
3) With changes by commit 02e6ed40387d10a17db48446e8856bbd644f04da and 85265a9b287ec860ee5c4b690bd913874d60f03a
```
Starting audit...
/home/andreiselkin/Downloads/temp/LocalizedMessage.java:106:12: More than 7 parameters (found 9).
/home/andreiselkin/Downloads/temp/LocalizedMessage.java:144:12: More than 7 parameters (found 8).
/home/andreiselkin/Downloads/temp/LocalizedMessage.java:175:12: More than 7 parameters (found 8).
Audit done.
Checkstyle ends with 3 errors.
```
4) With changes by commit 85265a9b287ec860ee5c4b690bd913874d60f03a and without changes by commit 02e6ed40387d10a17db48446e8856bbd644f04da
```
Starting audit...
/home/andreiselkin/Downloads/temp/LocalizedMessage.java:107:12: More than 7 parameters (found 9).
/home/andreiselkin/Downloads/temp/LocalizedMessage.java:145:12: More than 7 parameters (found 8).
/home/andreiselkin/Downloads/temp/LocalizedMessage.java:176:12: More than 7 parameters (found 8).
Audit done.
Checkstyle ends with 3 errors.
```
So, I could not reproduce the problem. For now ParameterNumber works fine on that example.

**Comment by MEZk** [comment]
Finally. I checkedout revision on 39ad631113210aadd4ab2a1e58514f55c7012328. I opened suppression.xml  and found 
```xml
<suppress id="paramNum"
  files="LocalizedMessage.java"
  lines="141,148,181,213"/>
```
It looks like that when commit 02e6ed40387d10a17db48446e8856bbd644f04da was done, it changed line numbers for suppressions that is why the build was failed.

**Comment by romani** [comment]
> ```<suppress id="paramNum"   files="LocalizedMessage.java"```

I do not see that config now.

So, what is a resolution ? if issue was due to problem in config that does not exists now - lets close issue

**Comment by MEZk** [comment]
@romani 
Your commit https://github.com/checkstyle/checkstyle/commit/02e6ed40387d10a17db48446e8856bbd644f04da deleted the line 56 and all line numbers in LocalizedMessage were changed. So suppressions did not work.  The problem was due to suppressions in suppression.xml which do not exist now, but, nevertheless, ParameterNumber check will rise violations on LocalizedMessage with default configuration (numbers of parameters is 7).

**Comment by romani** [comment]
That violations are ok, class is immutable, we need that amount of arguments.

But as we di not have suppressions for that class why we do not violations durung build ?

**Comment by MEZk** [comment]
Because property
```xml
 <property name="max" value="9"/>
```
for ParameterNumber is set to 9 in checkstyle_checks.xml:
 https://github.com/checkstyle/checkstyle/blob/master/config/checkstyle_checks.xml#L182

**Comment by romani** [comment]
it was added by me as workaround https://github.com/checkstyle/checkstyle/blame/master/config/checkstyle_checks.xml#L182

please update back to use default value, and put suppressions for that class of this check.

**Comment by romani** [comment]
merged
